src/open_inwoner/openzaak/tests/test_clients.py

```diff
@@ -1,28 +1,178 @@
+
 from django.test import TestCase
 
+import requests
+import requests_mock
 from zgw_consumers.constants import APITypes
-from zgw_consumers.models import Service
 
 from open_inwoner.openzaak.clients import (
+    MultiZgwClientProxy,
+    MultiZgwClientProxyResult,
+    ZgwClientResponse,
     build_catalogi_client,
+    build_catalogi_clients,
     build_documenten_client,
+    build_documenten_clients,
     build_forms_client,
+    build_forms_clients,
     build_zaken_client,
+    build_zaken_clients,
 )
+from open_inwoner.openzaak.exceptions import MultiZgwClientProxyError
 from open_inwoner.openzaak.tests.factories import ZGWApiGroupConfigFactory
 
 
 class ClientFactoryTestCase(TestCase):
     def setUp(self):
-        ZGWApiGroupConfigFactory()
+        self.api_groups = [
+            ZGWApiGroupConfigFactory(name="Default API"),
+            ZGWApiGroupConfigFactory(name="Second API"),
+        ]
 
     def test_originating_service_is_persisted_on_client(self):
-        for factory, api_type in (
-            (build_forms_client, APITypes.orc),
-            (build_zaken_client, APITypes.zrc),
-            (build_documenten_client, APITypes.drc),
-            (build_catalogi_client, APITypes.ztc),
+        for factory, api_type, api_group_field in (
+            (build_forms_client, APITypes.orc, "form_service"),
+            (build_zaken_client, APITypes.zrc, "zrc_service"),
+            (build_documenten_client, APITypes.drc, "drc_service"),
+            (build_catalogi_client, APITypes.ztc, "ztc_service"),
         ):
             client = factory()
-            self.assertIsInstance(client.configured_from, Service)
+            self.assertEqual(
+                client.configured_from, getattr(self.api_groups[0], api_group_field)
+            )
             self.assertEqual(client.configured_from.api_type, api_type)
+
+    def test_originating_service_is_persisted_on_all_clients(self):
+        for factory, api_type, api_group_field in (
+            (build_forms_clients, APITypes.orc, "form_service"),
+            (build_zaken_clients, APITypes.zrc, "zrc_service"),
+            (build_documenten_clients, APITypes.drc, "drc_service"),
+            (build_catalogi_clients, APITypes.ztc, "ztc_service"),
+        ):
+            clients = factory()
+            for i, client in enumerate(clients):
+                self.assertEqual(
+                    client.configured_from, getattr(self.api_groups[i], api_group_field)
+                )
+                self.assertEqual(client.configured_from.api_type, api_type)
+
+
+@requests_mock.Mocker()
+class MultiZgwClientProxyTests(PlainTestCase):
+    class SimpleClient:
+        def __init__(self, url):
+            self.url = url
+
+        def fetch_rows(self):
+            resp = requests.get(self.url)
+            return resp.json()
+
+    def setUp(self):
+        self.a_url = "http://foo/bar/rows"
+        self.another_url = "http://bar/baz/rows"
```

Comment:
```suggestion
        self.url_2 = "http://bar/baz/rows"
```
If it's not possible to use descriptive variable names (and I can't think of anything here), I'd prefer indexed names.